providers/base/bin/network_device_info.py

```diff
@@ -99,7 +97,13 @@ def get_ipv6_address(interface):
         cmd = ['/sbin/ip', '-6', '-o', 'addr', 'show', 'dev', interface,
                'scope', 'link']
         proc = check_output(cmd, universal_newlines=True)
-        return proc.split()[3].strip()
+        try:
+            ipv6_addr = proc.split()[3].strip()
+        except Exception as e:
```

Comment:
It's not a good practice to catch the general `Exception`. You can read more about this [here](https://www.learnpython.dev/03-intermediate-python/40-exceptions/70-best-practices/#don-t-catch-exception) and [here](https://python.land/how-not-to-handle-exceptions).

Check what specific exception is thrown when `proc` is not what the script expects, and handle this one.

Reply:
Yes, this did come to my mind before, I thought there might be some cases I miss if I specify OSError, so I just follow the code of get_ipv4_address().

So would it will be better if we modify the both functions?

Reply:
I modified get_ipv4_addr() & get_ipv6_addr() to check the specific error